### ISSUE TITLE
Install title from USB - Driver issue fix

### ISSUE THREAD

**Issue by xXS4ALLx** [rashevskyv/dbi#52]
For those on Windows 10 with all the latest updates/patches/etc...

If your having issues with the "**Install title from USB**" where Windows doesn't detect your Switch, it could be that the drivers required to connect to your Switch is messed up.

I just recently started using dbi and the ability to install over USB from the PC was a great feature, saving time to having to copy the files to the sd card and then install.

I couldn't get this feature to work since my device manager on Windows, shows a "**!**" telling me no drivers installed.

However the "**Run MTP responder**" works fine, and my Windows can see the Switch, I tried to install using this method but from what I can see, it first copies the files to the Switch, then installs, whereas the "**Install title from USB**" feature installs on the fly, so it doesn't copy the files first, then install.

Long story short, user by the name of **arabtm** over on **GBATemp** found the solution **[HERE](https://gbatemp.net/threads/dbi-usb-install-doesnt-work-anymore-on-2-out-of-3-usb-port.589818/#post-9499180)**.

You basically need to install the **tinfoil drivers** and **reg** file then reboot, this fixed my issue, download links to the drivers and reg file are provided in the GBATemp post above.

Hope this helps others in similar situation.

**Comment by duckbill007** [comment]
MTP way do not copy file first. It installs on the fly.
Instal process for backend and MTP is totally the same.

**Comment by xXS4ALLx** [comment]
Really? I must have seen what was going on the Switch screen wrong then?
What I thought I saw was files being copied first, then installed, then the file deleted.

So its basically the same them? If so, then, I much prefer the MTP method.

**Comment by duckbill007** [comment]
It is totally the same. The only diferense is: in backend method DBI asks for next file chunk, and in MTP method PC pushes file stream. In both cases file threated as stream, it'scontents on the fly copied to placeholders, after that old title deleted (if present on system), after that placeholders renamed to correct content names. Data read wform source and write to SD (NAND) only once.

**Comment by xXS4ALLx** [comment]
Great, thanks for the explanation, much appreciated, Im sure others will find this info useful.

Closing this comment.
